enterprise/backend/src/metabase_enterprise/gsheets/api.clj

```diff
@@ -204,116 +190,130 @@
+                   :created-at     created-at
+                   :gdrive/conn-id (-> response :body :id)
+                   :created-by-id  created-by-id})
+        (analytics/inc! :metabase-gsheets/connection-creation-began))
       (do
         (reset-gsheets-status!)
         (error-response-in-body
          (tru "Unable to setup drive folder sync.\nPlease check that the folder is shared with the proper service account email and sharing permissions."))))))
 
-(defn- sync-complete? [{:keys [status last-dwh-sync last-gdrive-conn-sync]}]
-  (and (= status "active") ;; HM says the connection is active
-       last-dwh-sync ;; make sure it's not nil
-       last-gdrive-conn-sync ;; make sure it's not nil
-       ;; We finished a sync of the dwh from metabase After the HM conn was synced:
-       (t/after? (t/instant last-dwh-sync) (t/instant last-gdrive-conn-sync))))
-
-(defn- get-last-mb-dwh-sync-time []
-  (t2/select-one-fn :ended_at :model/TaskHistory
-                    :db_id (t2/select-one-fn :id [:model/Database :id] :is_attached_dwh true)
-                    :task "sync"
-                    :status :success
-                    {:order-by [[:ended_at :desc]]}))
-
-(def ^:dynamic ^:private *folder-setup-timeout-seconds*
-  "We want to avoid polling forever, even if harbormaster never finishes the sync, so if the sync does not happen during
-  this window, we'll error out, reset the gsheets status, and suggest trying again."
-  (* 10 60))
+(defn- gsheets-safe
+  "Return the gsheets setting, or fetch it from the db if it's not in the cache."
+  []
+  (or (gsheets)
+      (do (log/warn "CACHE MISS ON GSHEETS")
+          (some-> (t2/select-one :model/Setting :key "gsheets")
+                  :value
+                  json/decode+kw))))
 
 (defn- handle-get-folder [attached-dwh]
-  (let [conn-id (or (:gdrive/conn-id (gsheets))
-                    (do (log/warn "CACHE MISS ON GSHEETS")
-                        (some-> (t2/select-one :model/Setting :key "gsheets")
-                                :value
-                                json/decode+kw
-                                :gdrive/conn-id)))
-        [sstatus {conn :body}] (try (hm-get-gdrive-conn conn-id)
-                                    ;; missing id:
-                                    (catch Exception _
-                                      (reset-gsheets-status!)
-                                      (error-response-in-body
-                                       (tru "Unable to find google drive connection, please try again.")
-                                       {:conn-id conn-id})))]
-    (if (= :ok sstatus)
-      (let [{:keys [status] last-gdrive-conn-sync :last-sync-at
-             :as   _gdrive-conn} (normalize-gdrive-conn conn)
-            last-dwh-sync       (get-last-mb-dwh-sync-time)]
+  (let [saved-setting (gsheets-safe)
+        conn-id (:gdrive/conn-id saved-setting)
+        hm-response (if (empty? saved-setting)
+                      nil
```

Comment:
This would help someone understand what shape `hm-get-gdrive-conn` will return, and makes more sense during the destructuring on  line 242.

```suggestion
                      [:error nil]
```